tests/test.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash -e
+
+function finish {
+  echo 'Removing test environment'
+  echo '---'
+  docker-compose down -v
+}
+trap finish EXIT
+finish
+
+# normalises project name by filtering non alphanumeric characters and transforming to lowercase
+declare -x COMPOSE_PROJECT_NAME
+COMPOSE_PROJECT_NAME=$(echo ${BUILD_TAG:-"ansible-plugin-testing"} | sed -e 's/[^[:alnum:]]//g' | tr '[:upper:]' '[:lower:]')
+
+declare -x ANSIBLE_MASTER_AUTHN_API_KEY=''
+declare -x CONJUR_ADMIN_AUTHN_API_KEY=''
+declare -x ANSIBLE_CONJUR_CERT_FILE=''
+declare conjur_cid=''
+declare cli_cid=''
+declare conjur_proxy_cid=''
+declare ansible_cid=''
+
+function main() {
+
+  docker-compose up -d --build conjur conjur_proxy conjur_cli
+
+  conjur_cid=$(docker-compose ps -q conjur)
+  conjur_proxy_cid=$(docker-compose ps -q conjur_proxy)
+  cli_cid=$(docker-compose ps -q conjur_cli)
+
+  echo "Waiting for Conjur server to come up"
+  wait_for_conjur
+
+  echo "Fetching SSL certs"
+  fetch_ssl_certs
+
+  echo "Fetching admin API key"
+  CONJUR_ADMIN_AUTHN_API_KEY=$(docker exec "${conjur_cid}" conjurctl role retrieve-key cucumber:user:admin)
+
+  echo "Recreating conjur CLI with admin credentials"
+  docker-compose up -d conjur_cli
+  cli_cid=$(docker-compose ps -q conjur_cli)
+
+  echo "Configuring Conjur via CLI"
+  setup_conjur
+
+  echo "Fetching Ansible master host credentials"
+  ANSIBLE_MASTER_AUTHN_API_KEY=$(docker exec "${cli_cid}" conjur host rotate_api_key --host ansible/ansible-master)
+  ANSIBLE_CONJUR_CERT_FILE='/cyberark/tests/conjur.pem'
+
+  echo "Preparing Ansible for test run"
+  docker-compose up -d  --build ansible
+  ansible_cid=$(docker-compose ps -q ansible)
+
+  echo "Running tests"
+  run_test_cases
+}
+
+function wait_for_conjur {
+  docker exec "${cli_cid}" bash -c '
+    for i in $(seq 20); do
+      curl -o /dev/null -fs -X OPTIONS ${CONJUR_APPLIANCE_URL} > /dev/null && echo "ready" && break
+      echo -n "."
+      sleep 2
+    done
+  '
```

Comment:
rather than propagating this, can we use `conjurctl wait`? here is an example: https://github.com/cyberark/conjur-service-broker/blob/master/ci/configure_v5.sh#L3

Reply:
great, i didn't know that option.